### PR TITLE
[IMP] web: add support for custom handlers for ir.actions.report

### PR DESCRIPTION
**Pull request by sdegueldre**

### CODE REVIEW

**Comment by ged-odoo** [comment]
missing test!

File: addons/web/static/src/webclient/actions/action_service.js [hunk 1]
```diff
@@ -942,7 +949,7 @@ function makeActionManager(env) {
                 }
                 return _executeClientAction(action, options);
             case "ir.actions.report":
-                return _executeReportAction(action, options);
+                return _executeReportAction(action, options, env);
```
Comment (aab-odoo) [inline addons/web/static/src/webclient/actions/action_service.js]: no need to pass it here I think

File: addons/web/static/tests/webclient/actions/report_action_tests.js [hunk 2]
```diff
@@ -274,4 +274,42 @@ QUnit.module("ActionManager", (hooks) => {
...
+            if (action.id === 7 && !customHandlerCalled) {
+                customHandlerCalled = true;
+                assert.step("calling custom handler");
+                return true;
+            }
+            assert.step("falling through to default handler");
+        });
+        await doAction(webClient, 7, { onClose: () => assert.step("never called") });
+        assert.step("first doAction finished");
+        await doAction(webClient, 7, { onClose: () => assert.step("on_close") });
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/actions/report_action_tests.js]: I think I wouldn't have tested the onClose stuff

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo retry

**Pull request closed without merging**